classes/test.php

```diff
@@ -246,13 +246,14 @@ public function getTestedClassName()
         //
         // The constant and function mock engines, and the code coverage
         // scores will not work.
-        if ($this instanceof PHPUnit\Framework\TestCase) {
+        if (true === $this->breakRelationBetweenTestSuiteAndSUT &&
+            $this instanceof PHPUnit\Framework\TestCase) {
             return 'StdClass';
         }
 
         $testedClassName = parent::getTestedClassName();
 
-        return preg_replace('/test$/i', '', $testedClassName);
+        return preg_replace('/(?<!\\\)test$/i', '', $testedClassName);
```

Comment:
We don't want to remove `test` if the classname is `test`: We want to remove the `test` class suffix only.